test/diff_details_superbrut-net.html

```diff
@@ -0,0 +1,81 @@
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta charset="utf-8">
+		<meta http-equiv="X-UA-Compatible" content="IE=edge">
+		<title>Simulateur de coût d'embauche</title>
+		<style>
+			body {
+				font-family: 'Helvetica Neue', Arial, Helvetica, sans-serif;
+			}
+
+			/* Customize the widget's style */
+			body .SGMAPembauche {
+				padding-top: 3em;
+				margin: auto;
+			}
+		</style>
+		<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.32/browser.js"></script>
+	</head>
+
+	<body>
+		<script type="text/javascript" src="../dist/cout-embauche-widget.js"></script>
+		<noscript>
+			Le simulateur de coût d'embauche n'est accessible qu'avec JavaScript activé.
+			Pour l'utiliser, suivez les <a href="http://www.enable-javascript.com/fr/">instructions pour activer JavaScript dans votre navigateur web</a>.
+		</noscript>
+		<script type="text/babel">
+			// cotisations that we knowingly do not display in the details section
+			var hiddenCotisations = [ 'contribution_developpement_apprentissage' ]
+			var form = document.querySelector('.SGMAPembauche #input form')
+
+			// request their value as a result of the API call .
+			// They could later be added to the recomputation.
+			// var newAction = form.getAttribute('action') + '+' + hiddenCotisations.join('+')
+			// form.setAttribute('action', newAction)
+
+			// remake the open fisca request
+			form.querySelector('input[name="salaire_de_base"]').value = 1458
+
+			setTimeout(() => {
+				var salaireBrut = +document.querySelector('#salaire').value
```

Comment:
For several reasons.

1. Contributability: unary operators such as `+` or `~` are “tricks” that are not very common, and that are hard to search for on standard references and search engines since they are special characters. Developers that have not already encountered them have a hard time understanding what it does exactly. An explicit cast is clearer.
2. Readability: it is very easy to miss a unary operator when reading code, especially when it is right after an assignment. We are so used to reading `+=` that this `= +` is easy to misread, or to miss altogether.
3. Clarity: the precedence rules for unary operators are not intuitive, and are an easy source of hard-to-track bugs. Pathological example: `+ '1' + '2'`. Is it `3` or `'12'`? Now try figuring this out with long property paths. An explicit cast, OTOH, with its parentheses, makes it obvious what is cast.